transcom/mymove
# B 19531 int

## [B-19531](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19531)

## Summary

This task has been repurposed since it's original creation in accordance with the fluctuation of DTOD changes. It is now a follow-on to [B-20757](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20757) (see [int pr 1](https://github.com/transcom/mymove/pull/13789), [int pr 2](https://github.com/transcom/mymove/pull/13859) ), which takes the stopgap error message 20757 and also introduces it in the Prime Simulator for updating shipments.

Testing 20757 in our integration/loadtest environment has been tricky due to the current dtod implementation in that environment. This issue will likely hold up this ticket as well, but it can be tested on local environments with a little setup.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. First before testing, navigate to pkg/route/dtod_zip5_distance.go and add `distanceFloat = -1` on line 124. This will simulate a dtod outage and allow us to continue testing. This is the step that makes testing tricky on other environments with their dtod setups.
2. Once this has been completed, log in as a prime simulator user and navigate to any move. I've been using move code PARAMS to test for HHGs.
3. Select "Update Shipment" for any shipment. 
4. Change any values, such as "Destination Type" at the bottom, and click update.
5. After a pause, the error should appear at the top with the correct message.
6. That's HHG's sorted. PPMs are a little different, but to test them, you must log in as a customer and make a new move.
7. Add a PPM Shipment, and set a duty location to NS Mayport. This is our location for Prime-counseled PPMs.
8. Submit the shipment with any data.
9. Log in as a TOO. Fill out orders info and approve to send the shipment to Prime.
10. Log in as a prime, find your prime-counseled PPM.
11. Update the shipment.
12. Under either zip code, you'll need to provide a new zip code to hit the DTOD section of code. If you're simulating the DTOD outage still, any zip code will do. Otherwise, 23471 is a good zip code that fails as it's a PO box. 
13. After updating the zipcode, click submit, and the error message should appear.

### Frontend


## Screenshots
<img width="1537" alt="image" src="https://github.com/user-attachments/assets/785ed5ea-f058-4898-b431-f4fab8f6be7a">

I don't know that ANDI is required for prime sim, but either way it passes:
<img width="1537" alt="image" src="https://github.com/user-attachments/assets/b990eb47-4b05-4c2f-8fd7-52fd7cf3e8fd">
